frontend/src/views/AddAnimals/ModalComponent.tsx

```diff
@@ -0,0 +1,311 @@
+/**
```

Comment:
We're going to be adding at least two other animal types. I don't want this file to get too long. What about making a wrapper component and then insert the seperate animal components in?
Like I said, the api calls get cached, so we won't make redundant server calls.

Reply:
Followed this suggestion and now there is a separate modal component, with inserts of other separate animal form components to keep file sizes down.